Makefile

```diff
@@ -26,8 +42,8 @@ install_deps:
 		echo "Poetry is not installed. Please install it from https://python-poetry.org/docs/#installation"; \
 		exit 1; \
 	fi
-	poetry install
-	npm install
+	@make install_python_deps
```

Comment:
is there a reason why these are run via `@make` rather than being a dependency of the task? Just looks a bit strange like this

Also I think you can drop the checks for npm and poetry above as they're included twice now.